ecommerce/src/Relecloud.Web/Startup.cs

```diff
@@ -97,13 +93,18 @@ public void ConfigureServices(IServiceCollection services)
             }
 
             // Add authentication if configured.
-            if (!string.IsNullOrWhiteSpace(authenticationConfigurationSection.GetValue<string>("ClientId")))
+            if (!string.IsNullOrWhiteSpace(Configuration.GetValue<string>("AzureAdB2C:ClientId")))
             {
-                services.Configure<AuthenticationConfiguration>(authenticationConfigurationSection);
-                services.AddAuthentication(CookieAuthenticationDefaults.AuthenticationScheme)
-                    .AddCookie(options => options.LoginPath = new PathString("/account/signin"))
-                    .AddOpenIdConnect(/* The options are more complex and are therefore configured separately using IConfigureOptions for clarity */);
-                services.AddSingleton<IConfigureOptions<OpenIdConnectOptions>, OpenIdConnectOptionsSetup>();
+                AddAzureAdB2cServices(services);
+            }
+
+            if (string.IsNullOrWhiteSpace(applicationInsightsConnectionString))
```

Comment:
Not sure we should throw if this isn't set. Generally telemetry should work when it can and not mess other stuff up when it can't.

Reply:
mistakenly thought i had to throw because i didn't have an alternative.